### ISSUE TITLE
How to use with Authentication

### ISSUE THREAD

**Issue by spawnia** [DavidDuwaer/Coloquent#29]
Hi, i want to use Coloquent in my project - the syntax and ease of use look really nice. 

My API requires Authentication via an Access Token sent in the Header. While looking through the source code of *Model*, i noticed that you are not exposing Axios options to the child class.

Am i missing something, is there an easy way to do this? If not, are there plans of exposing such configuration?

Thank you

**Comment by DavidDuwaer** [comment]
I'll release a fix tonight, it will basically come down to that you can inject your own AxiosInstance with a method setApiClient on Model.

**Comment by spawnia** [comment]
Really cool, i will take a look once you have it and happily test it!

**Comment by DavidDuwaer** [comment]
I've addressed the issue.

To protect future backward-compatibility, I first had to isolate Coloquent from the HTTP client it uses by wrapping the HTTP client (and its promise, and its response) in interfaces. You can now obtain the AxiosInstance from Model by doing
`instanceOfModel.getHttpClient().getImplementingClient()`. You can set its configuration in that object's `.default` property.

**Comment by DavidDuwaer** [comment]
@spawnia @kuzyk Let me know if this solves your problem!

**Comment by spawnia** [comment]
Thank you for taking the time to work on this. Your solution does work, and it offers a lot of flexiblity.

However, i think authentication via JWT or similar mechanisms is such a common use case and should probably be a first-class configuration option. It would be useful to have an example configuration, probably via a shared Model class, overriding the default configuration.

**Comment by andrewn** [comment]
I agree that it would be very useful to set the `Authorization` header (or some other custom headers) without needing to reimplement the `HttpClient` interface.

**Comment by DavidDuwaer** [comment]
(Update to the new version (v1.0.2), the exposed HttpClient was not static on the Model class, causing a new client instance to be used for every model instance; this is now fixed.)

You can set the HTTP client headers by getting the "implementing client" instance, and setting the proeprty there. Since Coloquent by default has Axios, you can do it like this:

```typescript
let axiosInstance: AxiosInstance = YourModel.getHttpClient().getImplementingClient();
axiosInstance.defaults.headers.authentication = 'someAuthenticationHeader5636rt3';
```

After setting this, the authorization header is set for all queries. We can make an API so you can set default headers directly on Model, I'm open to pull requests. Would be nice if the configuration methods are fluent as well (e.g. `Model.setDefaultHeader('foo', 'value').setDefaultHeader('bar', 'value')`). Also, do you need to be able to set it for specific requests? I can imagine a `.header` operator to be useful in a query.

**Comment by StephanMeijer** [comment]
Really easy to solve..

```
import axios from 'axios';
axios.defaults.headers.common['Authorization'] = 'Bearer asdasdasd'
```
